dandi/files/zarr.py

```diff
@@ -540,14 +541,38 @@ def mkzarr() -> str:
 def _upload_zarr_file(
     storage_session: RESTFullAPIClient, upload_url: str, item: UploadItem
 ) -> int:
-    with item.filepath.open("rb") as fp:
-        storage_session.put(
-            upload_url,
-            data=fp,
-            json_resp=False,
-            retry_if=_retry_zarr_file,
-            headers={"Content-MD5": item.base64_digest},
-        )
+    with ExitStack() as stack:
+        if os.environ.get("DANDI_DEVEL_INSTRUMENT_REQUESTS_SUPERLEN"):
```

Comment:
please add to CONTRIBUTING.md where we describe such variables

Reply:
We don't have a `CONTRIBUTING.md`.  Envvars are described in `DEVELOPMENT.md`.